packages/health/ios/Classes/SwiftHealthPlugin.swift

```diff
@@ -383,6 +416,74 @@ public class SwiftHealthPlugin: NSObject, FlutterPlugin {
         }
       })
   }
+    
+    func writeMeal(call: FlutterMethodCall, result: @escaping FlutterResult) throws {
+        guard let arguments = call.arguments as? NSDictionary,
+            let startTime = (arguments["startTime"] as? NSNumber),
+            let endTime = (arguments["endTime"] as? NSNumber),
+            let calories = (arguments["caloriesConsumed"] as? Double),
+            let carbs = (arguments["carbohydrates"] as? Double?) ?? 0,
+            let protein = (arguments["protein"] as? Double?) ?? 0,
+            let fat = (arguments["fatTotal"] as? Double?) ?? 0,
+            let name = (arguments["name"] as? String?)
+        else {
+            throw PluginError(message: "Invalid Arguments")
+        }
+        let dateFrom = Date(timeIntervalSince1970: startTime.doubleValue / 1000)
+        let dateTo = Date(timeIntervalSince1970: endTime.doubleValue / 1000)
+        
+        let hour = Calendar.current.component(.hour, from: dateFrom)
+        var mealType = "Snacks"
+        
+        if(hour > 5 && hour < 11) {
+            mealType = "Breakfast"
+        } else if(hour > 12 && hour < 15) {
+            mealType = "Lunch"
+        } else if(hour > 18 && hour < 22) {
+            mealType = "Dinner"
+        }
+        
+        var metadata = ["HKFoodMeal": "\(mealType)"]
+        
+        if(name != nil) {
+            metadata[HKMetadataKeyFoodType] = "\(name!)"
+        }
```

Comment:
```suggestion
var metadata = ["HKMetadataKeyFoodType": "\(mealType)"]
```

Reply:
We discovered a crash on iOS 16 and this fixes it.

Reply:
@hoffmatteo Please confirm if you don't need this fix. This caused a crash on our fork.

Reply:
I am not sure I understand, you are removing HKFoodMeal and replacing it with HKMetadataKeyFoodType? But then you lose the information of having both the name of the food and its mealtype right?

Reply:
I'm not sure either, the crash was a bit random and this seemed to fix it. If you can please test with some version of iOS 16, to be sure, otherwise if you're confident we can leave it like that, and if it's a problem fix it in a different pull request. Up to you 👍

Reply:
For now I would leave it like that, if there are any issues we can always fix it in the future!